eos-updater/apply.c

```diff
@@ -96,6 +96,66 @@ get_test_osname (void)
   return g_getenv ("EOS_UPDATER_TEST_UPDATER_OSTREE_OSNAME");
 }
 
+static gboolean
+update_remote_branches (OstreeRepo   *repo,
+                        const gchar  *refspec,
+                        GError      **error)
+{
+  g_autofree gchar *remote = NULL;
+  g_autofree gchar *ref = NULL;
+  g_autofree gchar *cur_branches = NULL;
+  g_autofree gchar *new_branches = NULL;
+
+  if (!ostree_parse_refspec (refspec, &remote, &ref, error))
+    return FALSE;
+  if (remote == NULL)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                   "Invalid refspec ‘%s’ in origin: did not contain a remote name",
+                   refspec);
+      return FALSE;
+    }
+
+  /* Update the remote branches option to include the origin ref.
+   * Ideally this would operate on a list of branches and add the new
+   * branch, but to keep things simple just set the option to the single
+   * origin ref. */
+  if (!ostree_repo_get_remote_option (repo, remote, "branches", NULL,
+                                      &cur_branches, error))
+    return FALSE;
+  new_branches = g_strdup_printf ("%s;", ref);
```

Comment:
Do we want to unconditionally overwrite all the old branches here?

Reply:
Only listing a single branch makes sense to me: the user is only ever going to be on a single OSTree branch apart from when they’ve just upgraded from (say) eos3.3 to eos3.4 and hence have one deployment from each. Even then, I can only see them wanting to pull eos3.4 going forwards.

Reply:
FWIW, I worked out how to merge the lists. If you're still interested in that, I can push my branch out.

The use case for multiple branches is more like a developer that's doing something very specific. Still, since the only use for the `branches` option is an `ostree pull` with no specified refs, it's very unlikely that just putting the single branch is the wrong thing to do.

Reply:
I don’t think it’s worth the churn, unless you think it’s worth it.

Reply:
Not really, but since I did the work I thought I'd offer :)